### PR TITLE
Feature/issue 224 create a reservation component

### LINKED ISSUE #224: Create a reservation component
At the moment, to be used inside the manager module.

This should include:
- A reservation list
- A reservation detail

### PR DESCRIPTION
**Pull request by CosNig**
# Feature/issue 224 create a reservation component

Updated reservation list component in order to display a list of reservations.
Updated detail component in order to display reservation details and tenant details.
Added commonTestingModule import to fix failed tests.

# Developer Checklist

- [ ] Updated documentation or README.md
- [ ] If adding new feature(s), added and ran unit tests
- [ ] If create new release, bumped version number
- [x] Ran `npm run style:fix` for code style enforcement
- [x] Ran `npm run lint:fix` for linting
- [ ] Ran `npm audit` to discover vulnerabilities

### CODE REVIEW

**Review by FanniFelfoldi — DISMISSED** [review]

File: src/app/modules/core/manager/reservation/components/details/reservation-details.component.ts [hunk 1]
```diff
@@ -1,12 +1,31 @@
...
+import { Observable } from 'rxjs'
 
 @Component({
   selector: 'cav-reservation-details',
   templateUrl: './reservation-details.component.html',
   styleUrls: ['./reservation-details.component.scss'],
 })
 export class ReservationDetailsComponent implements OnInit {
-  constructor() {}
+  tenantId: number
```
Comment (ludovico-zitelli) [inline src/app/modules/core/manager/reservation/components/details/reservation-details.component.ts]: Questa proprietà penso non serva più

File: src/app/modules/core/manager/reservation/components/details/reservation-details.component.ts [hunk 2]
```diff
@@ -1,12 +1,31 @@
...
+    private reservationService: ReservationService,
+    private tenantService: InMemoryTenantService
+  ) {}
 
-  ngOnInit(): void {}
+  ngOnInit(): void {
+    const id = parseInt(this.route.snapshot.paramMap.get('id'), 0)
+    this.reservationService.getById(id).subscribe((reservation) => {
+      this.tenant = this.tenantService.getById(reservation.tenantId)
+      return (this.reservation = reservation)
```
Comment (ludovico-zitelli) [inline src/app/modules/core/manager/reservation/components/details/reservation-details.component.ts]: Qui non è necessario mettere il return

File: src/app/modules/core/manager/reservation/components/details/reservation-details.component.html [hunk 3]
```diff
@@ -1 +1,39 @@
...
+    <mat-card-content>
+      <p>Reservation Id: {{reservation?.id}}</p>
+      <p>Property Id: {{reservation?.propertyId}}</p>
+      <p>Check-in: {{reservation?.checkIn}}</p>
+      <p>Check-out: {{reservation?.checkOut}}</p>
+      <p>Guests number: {{reservation?.guestNumber}}</p>
+    </mat-card-content>
+  </mat-card>
+
+  <div id='request'>
```
Comment (ludovico-zitelli) [inline src/app/modules/core/manager/reservation/components/details/reservation-details.component.html]: Questo id non penso ti serva

File: src/app/modules/core/manager/reservation/components/details/reservation-details.component.ts (hunk 1 above)
Comment (CosNig) [reply src/app/modules/core/manager/reservation/components/details/reservation-details.component.ts]: Sì, hai ragione.

**Review by FanniFelfoldi — DISMISSED** [review]

**Review by ludovico-zitelli — DISMISSED** [review]

**Review by pc-codingavision — APPROVED** [review]

**Pull request merged**